### PR TITLE
Remove using std::* statements in header files

### PR DESCRIPTION
**Pull request by rolandlo**
As wished by @Febbe on the TODO list, I removed using `std::` statements in all header files, and added `std::` accordingly throughout the header files. In the source files I added `std::` when there were only few occurences. Otherwise I added `using std::` in the source file.

### CODE REVIEW

File: src/util/XournalType.h [hunk 1]
```diff
@@ -26,7 +26,4 @@
 // Include string and vector everywhere
 #include <cstring>  // Todo: Use std::string_view if possible
 #include <string>
-#include <vector>
-
-using std::string;
-using std::vector;
+#include <vector>
```
Comment (Febbe) [inline src/util/XournalType.h:29]: Remove this header completely.

**Comment by Febbe** [comment]
LGTM

File: src/util/XournalType.h (hunk 1 above)
Comment (rolandlo) [reply src/util/XournalType.h]: Done. I suggest to wait for  #2339 to get merged.

Comment (Febbe) [reply src/util/XournalType.h]: Yes, this is a very invasive change, breaking the code of nearly every PR.

**Review by Febbe — APPROVED** [review]

**Comment by LittleHuba** [comment]
This should be the last commit making it into 1.1.0.

All PRs not part of the 1.1.0 release can then update their work without further delaying the release of 1.1.0.

**Comment by Technius** [comment]
This is pretty low priority imo. Let's leave this for after the release--we don't want to introduce random breakage from refactoring changes.

**Comment by rolandlo** [comment]
> This is pretty low priority imo. Let's leave this for after the release--we don't want to introduce random breakage from refactoring changes.

In my opinion it's time to merge this PR. Although this PR changes lots of files and may lead to some merge conflicts in other PR's, solving these merge conflicts is almost trivial (stick to your own PR and add `std::` where needed). I went through all changes once more and verified that the PR does nothing else than removing the XournalType header and `using std::` statements in header files, which is compensated by 
- adding `std::` to `string`, `vector`, `list`, `ofstream` at lots of places, 
- adding `using std::string` or `using std::vector` in some `.cpp` files, where it seemed advantageous.
-  adding header files includes where needed

 As long as it compiles this cannot have introduced any bugs.

So, if no objections are raised, I will merge this PR in 24 hours.

**Comment by Febbe** [comment]
Yes do it.

**Pull request merged**